pkg/util/selector_test.go

```diff
@@ -0,0 +1,854 @@
+				rs: policyv1alpha1.ResourceSelector{
+					APIVersion: "v1",
+					Kind:       "Pod",
+					Name:       "unmatched",
+				},
+			},
+			want: PriorityMisMatch,
+		},
+		{
+			name: "[case 2] name matched, labels unset",
+			args: args{
+				rs: policyv1alpha1.ResourceSelector{
+					APIVersion: "v1",
+					Kind:       "Pod",
+					Name:       "test",
+				},
+			},
+			want: PriorityMatchName,
+		},
+		{
+			name: "[case 3] name unset, labels not matched",
+			args: args{
+				rs: policyv1alpha1.ResourceSelector{
+					APIVersion: "v1",
+					Kind:       "Pod",
+					LabelSelector: &metav1.LabelSelector{
+						MatchLabels: map[string]string{
+							"foo": "unmatched",
+						},
+					},
+				},
+			},
+			want: PriorityMisMatch,
+		},
+		{
+			name: "[case 3] name unset, labels matched",
+			args: args{
+				rs: policyv1alpha1.ResourceSelector{
+					APIVersion: "v1",
+					Kind:       "Pod",
+					LabelSelector: &metav1.LabelSelector{
+						MatchLabels: map[string]string{
+							"foo": "bar",
+						},
+					},
+				},
+			},
+			want: PriorityMatchLabelSelector,
+		},
+		{
+			name: "[case 4] name and labels unset, match all",
+			args: args{
+				rs: policyv1alpha1.ResourceSelector{
+					APIVersion: "v1",
+					Kind:       "Pod",
+				},
+			},
+			want: PriorityMatchAll,
+		},
+		{
+			name: "[case 4] labels error",
+			args: args{
+				rs: policyv1alpha1.ResourceSelector{
+					APIVersion: "v1",
+					Kind:       "Pod",
+					LabelSelector: &metav1.LabelSelector{
+						MatchLabels: map[string]string{
+							"": "bar",
+						},
+					},
+				},
+			},
+			want: PriorityMisMatch,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := ResourceSelectorPriority(resource, tt.args.rs); got != tt.want {
+				t.Errorf("ResourceSelectorPriority() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestClusterMatches(t *testing.T) {
+	cluster := &clusterv1alpha1.Cluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "cluster1",
+			Labels: map[string]string{
+				"foo": "bar",
+			},
+		},
+		Spec: clusterv1alpha1.ClusterSpec{
+			Zone: "zone1",
+		},
+	}
+
+	type args struct {
+		affinity policyv1alpha1.ClusterAffinity
+	}
+	tests := []struct {
+		name string
+		args args
+		want bool
+	}{
+		{
+			name: "cluster excluded",
+			args: args{
+				affinity: policyv1alpha1.ClusterAffinity{
+					ExcludeClusters: []string{cluster.Name},
+				},
+			},
+			want: false,
+		},
+		{
+			name: "[case 1] labels not matched",
+			args: args{
+				affinity: policyv1alpha1.ClusterAffinity{
+					LabelSelector: &metav1.LabelSelector{
+						MatchLabels: map[string]string{"foo": "unmatched"},
+					},
+					ClusterNames: []string{},
+					FieldSelector: &policyv1alpha1.FieldSelector{
+						MatchExpressions: []corev1.NodeSelectorRequirement{
+							{Key: ZoneField, Operator: corev1.NodeSelectorOpExists},
+						},
+					},
+				},
+			},
+			want: false,
+		},
+		{
+			name: "[case 1] clusterNames not matched",
```

Comment:
We don't need to add a case number to each case.

Reply:
The `case x` means the scenes describe in
https://github.com/karmada-io/karmada/blob/a795d147673cb8afea34a59839e6e216555fd1b0/pkg/util/selector.go#L84-L93